src/hooks/coreSDK/safeCoreSDK.ts

```diff
@@ -56,11 +64,11 @@ export const initSafeSDK = async ({
   if (!isValidMasterCopy(implementationVersionState)) {
     const masterCopy = implementation
 
-    const safeL1Deployment = getSafeSingletonDeployment({ network: chainId, version: safeVersion })
-    const safeL2Deployment = getSafeL2SingletonDeployment({ network: chainId, version: safeVersion })
+    const safeL1Deployment = getSafeSingletonDeployments({ network: chainId, version: safeVersion })
+    const safeL2Deployment = getSafeL2SingletonDeployments({ network: chainId, version: safeVersion })
 
-    isL1SafeSingleton = masterCopy === safeL1Deployment?.networkAddresses[chainId]
-    const isL2SafeMasterCopy = masterCopy === safeL2Deployment?.networkAddresses[chainId]
+    isL1SafeSingleton = isInDeployments(masterCopy, safeL1Deployment?.networkAddresses[chainId])
+    const isL2SafeMasterCopy = isInDeployments(masterCopy, safeL2Deployment?.networkAddresses[chainId])
```

Comment:
I'm fine with the change, but if you have don't have the context on the L1/L2 deployment situation I doubt that anyone would be able to figure out why we do all this magic :D

Reply:
Our deployment structure changed in general. Now there can be multiple contract deployments per contract. So this would be correct regardless of this case in the future.